### PR TITLE
BlockManager: Change the order of validators to frozen UTXO

### PR DESCRIPTION
**Pull request by MichaelKim20**
The order of validators in Agora was sorted according to their frozen UTXO.
I change the same as Agora.

### CODE REVIEW

File: tests/Utils.ts [hunk 1]
```diff
@@ -1429,17 +1429,6 @@ export class EnrollmentStorage {
...
-        const keys: PublicKey[] = [];
-        this.storage.forEach((value, key, mapObject) => {
-            keys.push(new PublicKey(key));
-        });
-        keys.sort((a, b) => {
-            return PublicKey.compare(a, b);
-        });
-        return keys;
-    }
-
```
Comment (MichaelKim20) [inline tests/Utils.ts:1442]: Remove it because it's not used.

File: tests/Utils.ts [hunk 2]
```diff
@@ -1545,8 +1534,10 @@ export class FrozenUTXOStorage {
         this.storage.set(address.toString(), data);
     }
 
-    public get(address: PublicKey): undefined | Hash {
-        return this.storage.get(address.toString());
+    public get(address: PublicKey): Hash {
+        const utxo = this.storage.get(address.toString());
+        if (utxo === undefined) throw new Error("Not found frozen utxo");
```
Comment (MichaelKim20) [inline tests/Utils.ts:1539]: Throw an error because frozen UTXO must exist.

File: tests/Utils.ts [hunk 3]
```diff
@@ -1642,10 +1633,12 @@ export class BlockManager {
...
         });
 
-        this.validators.set(
-            this.height,
-            iota(0, 6).map((m) => ValidatorKey.keys(m))
-        );
+        const validators = iota(0, 6).map((m) => ValidatorKey.keys(m));
+        validators.sort((a, b) => {
+            return Hash.compare(this.frozen_utxos.get(a.address), this.frozen_utxos.get(b.address));
+        });
```
Comment (MichaelKim20) [inline tests/Utils.ts:1639]: Sort frozen UTXO in ascending order.

File: tests/Utils.ts [hunk 4]
```diff
@@ -1739,7 +1732,16 @@ export class BlockManager {
...
         const bits = new BitMask(validators.length);
-        validators.forEach((validator, idx) => {
+
+        const new_validators = validators.map((m) => m);
+        new_validators.sort((prev, next) => {
+            return Hash.compare(
+                this.frozen_utxos.get(prev.validator.address),
+                this.frozen_utxos.get(next.validator.address)
+            );
+        });
```
Comment (MichaelKim20) [inline tests/Utils.ts:1742]: Sort frozen UTXO in ascending order.

File: tests/Utils.ts [hunk 5]
```diff
@@ -1877,11 +1878,7 @@ export class BlockManager {
         if (height === undefined) height = this.height;
         const validators = this.validators.get(height);
 
-        return validators
-            .map((m) => m.address)
-            .sort((a, b) => {
-                return PublicKey.compare(a, b);
-            });
```
Comment (MichaelKim20) [inline tests/Utils.ts:1884]: Remove the function of sorting because it is already sorted.

**Review by TrustHenry — APPROVED** [review]
Thank you 👍

File: tests/Congress.test.ts [hunk 6]
```diff
@@ -224,8 +226,8 @@ describe("Test for the addition of validators", () => {
             const validators = await boa_client.getAllValidators(block_manager.getLastBlockHeight());
             assert.strictEqual(validators.length, validators_simulation.length);
             assert.deepStrictEqual(
-                validators.map((m) => m.address.toString()).sort((a, b) => a.localeCompare(b)),
-                validators_simulation.map((m) => m.toString()).sort((a, b) => a.localeCompare(b))
+                validators.map((m) => m.address.toString()),
+                validators_simulation.map((m) => m.toString())
```
Comment (MichaelKim20) [inline tests/Congress.test.ts:230]: It's in the same order as Stoa.

File: tests/Congress.test.ts [hunk 7]
```diff
@@ -166,6 +165,10 @@ describe("Test for the addition of validators", () => {
         const validators_simulation = block_manager.getValidators(block_manager.getLastBlockHeight());
         const validators = await boa_client.getAllValidators(block_manager.getLastBlockHeight());
         assert.strictEqual(validators.length, validators_simulation.length);
+        assert.deepStrictEqual(
+            validators.map((m) => m.address.toString()),
+            validators_simulation.map((m) => m.toString())
+        );
```
Comment (MichaelKim20) [inline tests/Congress.test.ts:171]: It's in the same order as Stoa.

File: tests/Utils.ts [hunk 8]
```diff
@@ -1811,7 +1813,7 @@ export class BlockManager {
         });
 
         res.sort((a, b) => {
-            return PublicKey.compare(a.validator.address, b.validator.address);
+            return Hash.compare(this.frozen_utxos.get(a.validator.address), this.frozen_utxos.get(b.validator.address));
```
Comment (MichaelKim20) [inline tests/Utils.ts:1816]: Sort frozen UTXO in ascending order.

File: tests/Utils.ts [hunk 9]
```diff
@@ -1827,9 +1829,8 @@ export class BlockManager {
                 validators.push(elem.validator);
             }
         }
-
         validators.sort((a, b) => {
-            return SecretKey.compare(a.secret, b.secret);
+            return Hash.compare(this.frozen_utxos.get(a.address), this.frozen_utxos.get(b.address));
```
Comment (MichaelKim20) [inline tests/Utils.ts:1833]: Sort frozen UTXO in ascending order.

**Pull request merged**